src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ConnectSqlCmdCommand.cs

```diff
@@ -71,6 +71,7 @@ private DbConnection AttemptToEstablishCurConnection(SqlConnectionInfo ci)
                 connString += ";Pooling=false"; //turn off connection pooling (this is done in other tools so following the same pattern)
 
                 conn = new SqlConnection(connString);
+                (conn as SqlConnection).RetryLogicProvider = Connection.ReliableConnection.RetryPolicyUtils.ServerlessWaitRetryLogicProvider();
```

Comment:
An `as` without a check for null is almost always a mistake: either cast (if you know it's a SqlConnection) or check for null.

Reply:
Considering that at line 73 you new'ed the SqlConnection, I fail to see why you'd want the `as`. Could you explain?

Reply:
@Charles-Gagnon Yeah that's right, thought it was needed there for a reason, but if that's safe to do so, I'm going with the change.